Framework/API/src/FileProperty.cpp

```diff
@@ -396,5 +401,55 @@ std::string FileProperty::convertExtension(const std::string &filepath) const {
   fullpath.setExtension(ext);
   return fullpath.toString();
 }
+
+/** Expand user variables in file path.
+ *  On UNIX, ~ is replaced by the user's home directory, if found.
+ *  On Windows, a combination of HOMEPATH and HOMEDRIVE will be used.
+ *  If the path contains no user variables, or expansion fails, the path is
+ *  returned unchanged, for errors to be dealt with by the calling function
+ *  @param filepath The path to expand
+ *  @return The expanded path
+ */
+std::string FileProperty::expandUser(const std::string &filepath) const {
+  auto start = filepath.begin();
+
+  if (*start != '~') // No user variables in filepath
+    return filepath;
+
+  // Position of the first slash after the variable
+  auto nextSlash = find_if(start, filepath.end(), &isSlash);
+
+  // UNIX-style home variable (ie ~/blah)
+  if (std::distance(start, nextSlash) == 1) {
+    char *home;
+
+    if (!(home = std::getenv("HOME")))
+      if (!(home = std::getenv("USERPROFILE")))
+        if (!(home = std::getenv("HOMEPATH")))
```

Comment:
`HOMEPATH` doesn't seem to add the drive letter, and will fail if it reaches that point, it would probably be better to remove it.

Also as it currently is, even if the variable exists, but is empty, e.g. `HOME="";USERPROFILE="/home/username` the condition will still go through on the first check for `HOME` and return the empty string, which causes a failure. I am not sure if we want to account for that.

@martyngigg Do you know if there is a chance for having a declared `HOME` and/or `USERPROFILE` but with an empty value?